### PR TITLE
Save and load improvements 

### PR DESCRIPTION
**Pull request by Joshua12450**

### CODE REVIEW

**Review by NoahMKuzyk — CHANGES_REQUESTED** [review]
Interacting with shopkeeper softlocks the player

**Review by Betadrone — APPROVED** [review]

File: CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs [hunk 1]
```diff
@@ -94,6 +102,69 @@ public GameObject ScanForLowestHealthEnemy()
...
     }
+
+    public List<PlayerAbility> GetAbilities() => playerAbilities;
+
+    public void ChangeAbilties(int abilityone, int abilitytwo)
+    {
+        List<PlayerAbility> ability = new List<PlayerAbility>();
+        switch (abilityone)
+        {
+            case 1:
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs:113]: should make an enum for these cases for each ability type

File: CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs [hunk 2]
```diff
@@ -94,6 +102,69 @@ public GameObject ScanForLowestHealthEnemy()
...
+            case 0:
+               ability[0] = null;
+                break;
+            default:
+                Debug.Log(ability[0]);
+                Debug.Log("Ability one not found");
+                ability[0] = null;
+                break;
+        }
+        switch (abilitytwo)
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs:137]: should we just make a function for this switch to save from it being duplicated? takes in the ability as one param and the ability to set as the other?

File: CapstoneProject/Assets/Scripts/Core/Entity/Characters/Playable_Character/Player.cs [hunk 3]
```diff
@@ -63,27 +84,159 @@ protected override void InitialSetup()
...
+                CoinComponent.ChangeCoins(data.playerCoins);
+                KeyItemComponent.ChangeKeys(data.playerKeys);
+                HealthComponent.ChangeHealth(data.playerHealth);
+                var ability = AbilityComponent.playerAbilities;
+/*                List<PlayerAbility> ability = new List<PlayerAbility>()
+                {
+                    null,
+                    null
+                };*/
+                switch (data.playerAbilityOne)
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Entity/Characters/Playable_Character/Player.cs:171]: same here, could be a function

**Pull request merged**